samples/nrf9160/nrf_cloud_rest_cell_location/src/main.c

```diff
@@ -195,17 +211,49 @@ static void get_cell_info(void)
 	} else {
 		struct lte_lc_ncellmeas_params ncellmeas_params = {
 			.search_type = search_type,
+			.gci_count = CONFIG_REST_CELL_GCI_COUNT
 		};
 
-		LOG_INF("Requesting neighbor cell measurement...");
+		/* Set the result buffers */
 		cell_info.neighbor_cells = neighbor_cells;
+		cell_info.gci_cells = gci_cells;
+
+		LOG_INF("Requesting neighbor cell measurement");
 		err = lte_lc_neighbor_cell_measurement(&ncellmeas_params);
 		if (err) {
 			LOG_ERR("Failed to start neighbor cell measurement, error: %d", err);
+		} else {
+			LOG_INF("Waiting for measurement results...");
 		}
 	}
 }
 
+static void process_cell_pos_type_change(void)
+{
+	if (ready) {
+		/* Toggle active cell pos type */
+		if (active_cell_pos_type == LOCATION_TYPE_SINGLE_CELL) {
+			if (cur_rrc_mode == LTE_LC_RRC_MODE_IDLE) {
+				/* RRC is idle, switch to multi-cell type */
+				active_cell_pos_type = LOCATION_TYPE_MULTI_CELL;
+			} else if (rrc_idle_wait == false) {
+				LOG_INF("Waiting for RRC idle mode "
+					"before requesting neighbor cell measurement...");
+				/* Wait for RRC idle. If this function is called before
+				 * idle, the wait is cancelled and single-cell is used.
+				 */
+				rrc_idle_wait = true;
+				return;
+			}
+		} else {
+			active_cell_pos_type = LOCATION_TYPE_SINGLE_CELL;
+		}
+
+		rrc_idle_wait = false;
+		get_cell_info();
+	}
+}
```

Comment:
For me, a flatter structure is easier to follow, but I'll admit this is subjective:
```suggestion
static void process_cell_pos_type_change(void)
{
	if (!ready) {
		return;
	}
	
	/* Toggle active cell pos type */
	if (cur_rrc_mode == LTE_LC_RRC_MODE_IDLE) {
		/* RRC is idle, switch to multi-cell type */
		active_cell_pos_type = LOCATION_TYPE_MULTI_CELL;
		rrc_idle_wait = false;
		get_cell_info();
	} else if (rrc_idle_wait == false) {
		/* RRC is not idle, and we haven't started waiting yet.
		 * wait for RRC idle.
		 */
		LOG_INF("Waiting for RRC idle mode "
			"before requesting neighbor cell measurement...");
		rrc_idle_wait = true;
	} else {
		/* RRC is not idle, but this function has been called
		 * again after starting to wait for idle. Skip the wait
		 * and use single-cell.
		 */
		active_cell_pos_type = LOCATION_TYPE_SINGLE_CELL;
		rrc_idle_wait = false;
		get_cell_info();
	}
}
```

Reply:
Also, triple check that I actually left the functionality unchanged. I'm pretty sure I did, but check too just to be sure

Reply:
functionality is changed.
but i will do an early return if not ready to eliminate a nest.